### PR TITLE
Modify GitHub Actions to deploy image to ECR

### PR DESCRIPTION
**Pull request by nabeelamir-defra**
https://eaflood.atlassian.net/browse/IWTF-4240

Modify GitHub Actions to deploy image to ECR

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -84,3 +86,77 @@ jobs:
...
+            echo "| ---------- | ----- |"
+            echo "| created    | ${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.created'] }} |"
+            echo "| description| ${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.description'] }} |"
+            echo "| licenses   | ${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.licenses'] }} |"
+            echo "| revision   | ${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.revision'] }} |"
+            echo "| source     | ${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.source'] }} |"
+            echo "| title      | ${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.title'] }} |"
+            echo "| url        | ${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.url'] }} |"
+            echo "| version    | ${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.version'] }} |"
+          } >> $GITHUB_STEP_SUMMARY
```
Comment (jaucourt) [inline .github/workflows/build.yml:162]: nice!

File: Dockerfile [hunk 2]
```diff
@@ -1,22 +1,29 @@
...
-RUN npm install --ignore-scripts && npm cache clean --force > /dev/null 2>&1
-
-####################################################################################################################################
-# Build stage 2 - Using the distribution from stage 1, build the final docker image with a minimal number of layers.
-####################################################################################################################################
-FROM rod_catch_returns/base
-WORKDIR /app
-COPY --from=builder /app/ /app/
+RUN npm install pm2 -g --ignore-scripts > /dev/null 2>&1 \
+    && npm install --ignore-scripts && npm cache clean --force > /dev/null 2>&1
```
Comment (jaucourt) [inline Dockerfile:13]: why are you suppressing stdout and stderr for installation of pm2 and cache clean, but not for npm install? Don't we want all output, particularly whilst we're developing?

File: package.json [hunk 3]
```diff
@@ -16,10 +16,12 @@
...
-    "docker:build-dev": "TAG=dev PROJECT_DOCKERFILE=Dockerfile.dev docker compose -f docker/services.build.yml build",
-    "docker:services-dev": "TAG=dev docker stack deploy -c docker/services.dev.yml rcrs",
-    "docker:build": "PROJECT_DOCKERFILE=Dockerfile docker-compose -f docker/services.build.yml build",
-    "docker:services": "docker stack deploy -c docker/services.yml rcrs",
+    "docker:build-dev": "docker build -t rod_catch_returns/api_js_dev --target development .",
+    "docker:service-dev": "docker run -d --rm --env-file .env -p 5000:5000 -v ${PWD}/:/app --name rcr_api_js_dev rod_catch_returns/api_js_dev",
+    "docker:stop-dev": "docker container kill rcr_api_js_dev",
+    "docker:build": "docker build -t rod_catch_returns/api_js --target production .",
+    "docker:service": "docker run -d --rm --env-file .env -p 5000:5000 --name rcr_api_js rod_catch_returns/api_js",
+    "docker:stop": "docker container kill rcr_api_js",
```
Comment (jaucourt) [inline package.json:24]: Great that you've added these additional commands, but I think some use cases in the README might be handy

File: .github/workflows/build.yml (hunk 1 above)
Comment (nabeelamir-defra) [reply .github/workflows/build.yml:162]: This was copied over from https://github.com/DEFRA/cyltfr-service/blob/main/Dockerfile!

File: Dockerfile (hunk 2 above)
Comment (nabeelamir-defra) [reply Dockerfile]: I've removed `> /dev/null 2>&1` and the cache clean. I copied those over from rod-licensing. I guess the cache clean was there so when you do a docker build, the next time it runs does a fresh install of the dependencies. With it removed, the npm install runs a lot quicker!

File: package.json (hunk 3 above)
Comment (nabeelamir-defra) [reply package.json:24]: Good point, I've updated the README

**Review by jaucourt — APPROVED** [review]

**Pull request merged**